subprojects/core/src/main/java/org/gradle/internal/composite/CompositeBuildSettingsLoader.java

```diff
@@ -45,7 +45,10 @@ public CompositeBuildSettingsLoader(SettingsLoader delegate, ServiceRegistry bui
 
     @Override
     public SettingsInternal findAndLoadSettings(GradleInternal gradle) {
+        CompositeContextBuilder compositeContextBuilder = buildServices.get(CompositeContextBuilder.class);
```

Comment:
Yeah, the problem is that I tried to keep most of the composite build code in a separate subproject, but some services need to be injected into core services. This isn't new to this change, but I guess the code was moved. 

I think Adam added something recently to make this easier, at least by having a service that is injected with _all_ services of a type. I'll see if I can improve this.